### PR TITLE
stop initializing arrays to check for keys

### PR DESCRIPTION
**Pull request by nev-r**

### CODE REVIEW

**Comment by robojumper** [comment]
I assume this is for performance? Worth noting that `for...in` and `Object.keys()` have [slightly different semantics](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys#description) regarding prototypes. Probably not a concern for DIM's own code but that's all I can comment on this.

**Comment by nev-r** [comment]
not lost on be but it's being used for extremely rudimentary stuff in DIM and the unnecessary allocation was driving me crazy

File: src/app/bungie-api/destiny1-api.ts [hunk 1]
```diff
@@ -26,7 +27,7 @@ export async function getCharacters(platform: DestinyAccount) {
       `/D1/Platform/Destiny/${platform.originalPlatformType}/Account/${platform.membershipId}/`
     )
   );
-  if (!response || Object.keys(response.Response).length === 0) {
+  if (!response || !hasKeys(response.Response)) {
```
Comment (bhollis) [inline src/app/bungie-api/destiny1-api.ts:30]: I might either prefer `_.isEmpty` from Lodash, or to replace all existing `_.isEmpty` with this helper. `_.isEmpty` is certainly not as simple so might not be what we want: https://lodash.com/docs/4.17.15#isEmpty

**Review by bhollis — APPROVED** [review]

File: src/app/bungie-api/destiny1-api.ts (hunk 1 above)
Comment (nev-r) [reply src/app/bungie-api/destiny1-api.ts:30]: oh yeah that's ubiquitous in dim. forgot it existed

**Pull request closed without merging**